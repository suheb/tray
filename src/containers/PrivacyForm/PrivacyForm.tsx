import React, { useState, ChangeEvent } from "react";
import { Grid, Button, FormControlLabel, Checkbox } from "@material-ui/core";


interface PrivacyFormProps {
  formData?: { [field: string]: any }
  handleBack(): void
  handleSubmit(formName: string, formData: { [field: string]: any }): void
}

const PrivacyForm = (props: PrivacyFormProps) => {
  const initialFormData = {
    "firstPartyEmail": false,
    "thirdPartyEmail": false
  }
  const [formData, setFormData] = useState<{ [field: string]: any }>(props.formData || initialFormData)

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const value = event.target.checked
    const name = event.target.id

    // Set field value
    setFormData({
      ...formData,
      [name]: value
    })
  }

  const handleSubmit = () => {
    props.handleSubmit("privacy", formData);
  }
  return (
    <Grid
      container
      justify="flex-end"
      spacing={2}
    >
      <Grid
        item
        xs={12}
      >
        <FormControlLabel
          control={
            <Checkbox
              id="firstPartyEmail"
              color="primary"
              checked={formData["firstPartyEmail"] || false}
              onChange={handleChange}
            />
          }
          label="Receive updates about Tray.io products by email"
        />
      </Grid>

      <Grid
        item
        xs={12}
      >
        <FormControlLabel
          control={
            <Checkbox
              id="thirdPartyEmail"
              color="primary"
              checked={formData["thirdPartyEmail"] || false}
              onChange={handleChange}
            />
          }
          label="Receive communication by email for other products created by the Tray.io team"
        />
      </Grid>
      <Grid
        item
        xs={3}
      >
        <Button
          fullWidth
          variant="contained"
          color="primary"
          onClick={props.handleBack}
        >
          Back
        </Button>
      </Grid>
      <Grid
        item
        xs={3}
      >
        <Button
          fullWidth
          variant="contained"
          color="primary"
          onClick={handleSubmit}
        >
          Submit
        </Button>
      </Grid>
    </Grid>
  )
}

export default PrivacyForm;
