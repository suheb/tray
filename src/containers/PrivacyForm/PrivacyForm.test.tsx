import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

import PrivacyForm from './PrivacyForm';

describe("PrivacyForm", () => {
  it("renders without crashing", () => {
    const handleSubmit = jest.fn();
    const handleBack = jest.fn();
    const { asFragment } = render(<PrivacyForm handleBack={handleBack} handleSubmit={handleSubmit} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("does not submit if there are errors", () => {
    const handleSubmit = jest.fn();
    const handleBack = jest.fn();
    const formData = { "firstPartyEmail": false, "thirdPartyEmail": false }
    const { getByText } = render(<PrivacyForm handleBack={handleBack} handleSubmit={handleSubmit} />);
    userEvent.click(getByText(/submit/i));
    expect(handleSubmit).toBeCalledWith("privacy", formData);
  });

  it("does submits with correct values", async () => {
    const handleSubmit = jest.fn();
    const handleBack = jest.fn();
    const formData = { "firstPartyEmail": true, "thirdPartyEmail": false }
    const { getByText, getByLabelText } = render(<PrivacyForm handleBack={handleBack} handleSubmit={handleSubmit} />);
    await userEvent.click(getByLabelText(/receive updates/i));

    userEvent.click(getByText(/submit/i));
    expect(handleSubmit).toBeCalledWith("privacy", formData);
  });
});
