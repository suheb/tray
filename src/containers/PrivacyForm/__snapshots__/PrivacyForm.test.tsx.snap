// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyForm renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-116 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-119"
              data-indeterminate="false"
              id="firstPartyEmail"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Receive updates about Tray.io products by email
        </span>
      </label>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-116 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-119"
              data-indeterminate="false"
              id="thirdPartyEmail"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Receive communication by email for other products created by the Tray.io team
        </span>
      </label>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Back
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
